soyjs/exec.go

```diff
@@ -502,42 +503,96 @@ func (s *state) visitSwitch(node *ast.SwitchNode) {
 
 // TODO: {msg} node translation is unimplemented
 func (s *state) visitMsg(node *ast.MsgNode) {
-	var pluralNode, ok = node.Body.Children()[0].(*ast.MsgPluralNode)
-	if !ok {
-		s.visitMsgBody(node)
+	// If no bundle was provided, walk the message sub-nodes.
+	if s.options.Messages == nil {
+		s.visitMsgNode(node, node, nil)
 		return
 	}
 
-	s.jsln("switch (", pluralNode.Value, ") {")
+	// Look up the message in the bundle.
+	var msg = s.options.Messages.Message(node.ID)
+	if msg == nil {
+		s.visitMsgNode(node, node, nil)
+		return
+	}
+
+	s.visitMsgNode(node, node, msg.Parts)
+}
+
+func (s *state) visitMsgNode(n ast.ParentNode, root *ast.MsgNode, parts []soymsg.Part) {
+	for i, child := range n.Children() {
```

Comment:
might be worth adding a sanity check that parts is the same length as children?  (or is it nil if no bundle is provided?)